update checkbox page to let user know how to get rid of pad (#2865)
grommet/hpe-design-system: aries-site/src/pages/components/checkbox.mdx

Checked and unchecked states refer to when an individual CheckBox is checked or when all options from a group are checked.

### CheckBox outside of FormField

> By default, CheckBox has built-in `pad` to create space between a CheckBox and FormField border. When using CheckBox outside of a FormField, apply `pad="none"`.

Reviewer: ```suggestion
By default, CheckBox has built-in `pad` to create space between the CheckBox and FormField border. When using CheckBox outside of a FormField, apply `pad="none"`.
```

> The CheckBox has a built in `pad` to allow space when wrapped in a FormField. In order to have a stand alone CheckBox align with other FormFields in a Form users can explicatly 

Reviewer: ```suggestion
By default, CheckBox has built-in `pad` to create space between a CheckBox and FormField border. When using CheckBox outside of a FormField, apply `pad="none"`.
```

> set the `pad = "none"` in order to remove the pad. 

Reviewer: ```suggestion
```

> set the `pad: {{ vertical: 'xsmall' }}` in order to remove the horizontal pad. 

Reviewer: I thought we don't want any vertical pad outside of FormField, and that we're going to handle the hover treatment differently when outside of FormField

Reviewer: so should we say pad="none"? or say that pad can be controlled via `pad` prop and value set to accomodate the layout, but that mostly commonly it would be "none"?

Author: let me put this in draft, for now, sorry forgot about hover treatment I'll update once I chat with designers

Checked and unchecked states refer to when an individual CheckBox is checked or when all options from a group are checked.

> ### CheckBox outside of FormField

Reviewer: This feels like a candidate for an HPE specific eslint rule.
